File: main/java/com/hexaware/Cricketteammanagement/service/PlayerService.java
package com.hexaware.Cricketteammanagement.service;

import java.util.List;

import com.hexaware.Cricketteammanagement.Entity.Player;
import com.hexaware.Cricketteammanagement.dto.PlayerDTO;

public interface PlayerService {
   
	Player createPlayer(PlayerDTO playerDTO);
	List<PlayerDTO>getAllPlayers();
	PlayerDTO getPlayerById(Long id);
	PlayerDTO updatePlayer(Long id,PlayerDTO playerDTO);
	void deletePlayer(Long id);
}

File: coding challenge_Pavithra.C/src/app/player.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Player } from './models/player.model';

@Injectable({ providedIn: 'root' })
export class PlayerService {
  private baseUrl = 'http://localhost:8080/api/players';

  constructor(private http: HttpClient) {}

  getAllPlayers(): Observable<Player[]> {
    return this.http.get<Player[]>(this.baseUrl);
  }

  getPlayerById(id: number): Observable<Player> {
    return this.http.get<Player>(`${this.baseUrl}/${id}`);
  }

  addPlayer(player: Player): Observable<Player> {
    return this.http.post<Player>(this.baseUrl, player);
  }

  updatePlayer(player: Player): Observable<Player> {
    return this.http.put<Player>(`${this.baseUrl}/${player.id}`, player);
  }

  deletePlayer(id: number): Observable<string> {
    return this.http.delete(`${this.baseUrl}/${id}`, { responseType: 'text' });
  }
}
